app/src/main/java/com/github/sdp_begreen/begreen/viewModels/ConnectedUserViewModel.kt

```diff
@@ -1,16 +1,79 @@
 package com.github.sdp_begreen.begreen.viewModels
 
 import android.graphics.Bitmap
-import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.sdp_begreen.begreen.firebase.Auth
+import com.github.sdp_begreen.begreen.firebase.DB
 import com.github.sdp_begreen.begreen.models.User
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.*
+import org.koin.java.KoinJavaComponent.inject
 
 class ConnectedUserViewModel: ViewModel() {
-    val currentUser: MutableLiveData<User> by lazy {
-        MutableLiveData<User>()
+
+    private val db by inject<DB>(DB::class.java)
+    private val auth by inject<Auth>(Auth::class.java)
+
+    private val mutableCurrentUser = MutableStateFlow<User?>(null)
+    private val mutableCurrentUserProfilePicture = MutableStateFlow<Bitmap?>(null)
+
+    /**
+     * Flow that dynamically retrieve authenticated user from firebase upon any modification
+     */
+    private val userFromAuth = auth.getConnectedUserIds().map {
+        it?.let { db.getUser(it) }
+    }.onEach {
+        // upon new connection, start by resetting the profile picture to be sure to not keep an
+        // old picture, if the new authenticated person doesn't have a profile picture for example
+        mutableCurrentUserProfilePicture.value = null
+
+        // each time a new connection is done, directly fetch the user profile picture
+        // and update the currentUserProfilePicture value
+        it?.also { user ->
+            user.profilePictureMetadata?.let { photo ->
+                db.getUserProfilePicture(photo, user.id)
+            }?.also { photo -> mutableCurrentUserProfilePicture.value = photo }
+        }
     }
 
-    val currentUserProfilePicture: MutableLiveData<Bitmap> by lazy {
-        MutableLiveData<Bitmap>()
+    /**
+     * Merge both flows, mutableCurrentUser and userFromAuth, so that we always get
+     * the latest value either from the user updated one, or if an modification in the authenticated
+     * user arrives
+     */
+    @OptIn(FlowPreview::class)
+    val currentUser: StateFlow<User?> = flowOf(mutableCurrentUser, userFromAuth)
+        .flattenMerge()
+        .stateIn(viewModelScope, SharingStarted.Eagerly, null)
```

Comment:
The idea is that I want the flow `currentUser` to always emit the most up to date value of the connected user.

So the way I did it is that I had one `MutableStateFlow` that contains the most recent user that has been set by calling `setCurrentUser`. This method should be called whenever we modify some attributes of the current user, for example when we modify its username, then we should store it in the database, and call this method so that the view model will be up to date with the new `User`, and everyone that is collecting this flow will directly be updated with the correct values.

And then I had another flow of user that I called `userFromAuth` that will change upon changing in the authentication status.

But then as I want the most up to date values, I merged both those flows so that we can collect values from both of them. 

Here is an example of how the merge is done, basically as soon as a new value is emitted from either flow, they will be directly re-emitted from the merged flow. All flow are completely interleaved concurrently.

|--------o-------o----------o----------------------| : Flow 1
|----x----------------x-----------------------x-----| : Flow 2 
|----x---o------o---x------o---------------x-----| : Merged flow

And finally, the `stateIn` is there to transform a `cold flow` into a `hot flow` meaning that it keeps a single connection open, and re-emit the last known value upon new flow collection. Its value can also be retrieved by the use of the `.value` attribute.

Let me know if it still unclear, or if you have suggestion in the way of doing it.

We can also discuss it in person, it might be easier to explain

Reply:
Thank you so much for putting such effort in this message. =D